tools/templates/moar/Makefile.in

```diff
@@ -11,7 +11,7 @@ MOAR_PREFIX = @nfp(@moar::prefix@)@
 MOAR        = @nfpq(@moar::bindir@/moar@moar::exe@)@
 @bpv(NQP)@       = @nfpq(@m_nqp@)@
 @bpv(NQP_RR)@    = @bpm(NQP)@
-@bpv(RUN_RAKUDO)@ = $(MOAR) --libpath=@nfpq($(BASE_DIR)/blib)@ --libpath=@q(@bpm(NQP_LIBDIR)@)@ @bsm(RAKUDO)@ --rakudo-home=@nfpq($(BASE_DIR)/gen/build_rakudo_home)@
+@bpv(RUN_RAKUDO)@ = @shquot(@perl@)@ -e "\$$ENV{RAKUDO_HOME}='$(BASE_DIR)/gen/build_rakudo_home'; exec('@moar::bindir@/moar@moar::exe@', '--libpath=$(BASE_DIR)/blib', '--libpath=@bpm(NQP_LIBDIR)@', '@bsm(RAKUDO)@', @ARGV)" --
```

Comment:
I'm tearing apart between a few tasks and can't really focus. But looks ok for this approach, except for the assumption that `moar::bindir` won't contain a single quote. 

Though I'd extract the `-e` part into a separate script for readability. The script would be expanded from a template. It'd complicate the Makefile a bit but would allow to check the script content for errors.